### ISSUE TITLE
Lazy load on something like `exclude_ft`

### ISSUE THREAD

**Issue by cuducos** [wbthomason/packer.nvim#1131]
### Describe the feature

I love the `ft` config, as in `use {'FooBar/Baz', ft = 'bax'}` but sometimes I want a certain plugin to be available in every `ft` but one or two.

#### Use case
My use case is that a lot of treesitter + virualtext plugins do not make sense in languages that differ considerably in syntax from others, for example:

| Plugins that makes sense here | Might not make sense here |
| ---|---|
| C-style like Rust or Go | ML-style like Haskell or Elm |
| Blocks defined w/ open & close char/keywords, like JS or Ruby | Indented-based blocks like Python |

I would like to do, for example `use {'FooBar/Baz', exclude_ft = 'bax'}` and have this plugin working in every `ft` but `bax`.

##### Concrete example

[`haringsrob/nvim_context_vt`](https://github.com/haringsrob/nvim_context_vt) makes a lot of sense for almost all languages I use in my routine, but Python. In Python, it just pollutes the buffer. This plugin in particular has an exclusion list by `ft` and my idea came from there: why not bring this option to all Packer-managed plugins?

#### Possible implementation

I see that `ft` becomes `fts` and then we create auto commands from them:

https://github.com/wbthomason/packer.nvim/blob/d3ecfb4c8a33b46ea79356ef20b6a9e7789c86a2/lua/packer/compile.lua#L504-L514

We could do an extra check on that `for` loop to skip `ft` in `exlude_fts`, e.g.:

```lua
local function skip(ft)
	for _, val in ipairs(fts) do
		if val == ft then
			return true
		end
	end
	return false
end

for ft, names in pairs(fts) do
	if not skip(ft) then
		table.insert(
			ft_aucmds,
			-- etc ...
		)
	end
end
```

If that makes sense I am happy to open a PR, but as I have little experience with Lua I might need an bit of support and a few rounds of code reviews to make the PR production ready : )

**Comment by lewis6991** [comment]
In v2 this should be achievable via custom loaders (see #1162).

**Comment by max397574** [comment]
I think this would already possible with autocmd on `BufEnter` inside `setup` and `if not vim.tbl_contains({"filetype2","filetype1"},vim.bo[args.buf]) load....`

**Comment by cuducos** [comment]
100% agree, @lewis6991! Once #1162 is merged, I think we can close this issue. Thank you so much.

**Comment by cuducos** [comment]
Closed by #1162
